dhctl/cmd/dhctl/commands/converge.go

```diff
@@ -29,6 +31,31 @@ import (
 	"github.com/deckhouse/deckhouse/dhctl/pkg/terraform"
 )
 
+func getCacheIdentityFromKubeconfig(
+	kubeconfigPath string,
+	kubeconfigContext string,
+) (string, error) {
+	builder := strings.Builder{}
+	builder.WriteString("kubeconfig")
+
+	kubeconfigName := filepath.Base(kubeconfigPath)
+	if kubeconfigName == "." || kubeconfigName == ".." || kubeconfigName == "/" {
+		return "", fmt.Errorf("incorrect kubeconfig path")
+	}
+
+	builder.WriteString("-")
+	builder.WriteString(kubeconfigName)
```

Comment:
Cache identities will be the same for `/abc/kubeconfig` and `/def/kubeconfig`. Is it ok?

Reply:
Oops. We forget about it. Thanks